src/coap_net.c

```diff
@@ -3512,8 +3516,10 @@ handle_response(coap_context_t *context, coap_session_t *session,
                                                      rcvd->mid));
     if (ret == COAP_RESPONSE_FAIL && rcvd->type != COAP_MESSAGE_ACK) {
       coap_send_rst(session, rcvd);
+      session->last_con_handler_res = COAP_RESPONSE_FAIL;
     } else {
       coap_send_ack(session, rcvd);
+      session->last_con_handler_res = COAP_RESPONSE_OK;
     }
   } else {
     coap_send_ack(session, rcvd);
```

Comment:
You need to add in after this line
````
last_con_handler_res = COAP_RESPONSE_OK;
````
as it is possible that context->response_handler (new line 1512) is now not set, but previously could have been set (but very unlikely).